specification/app/resource-manager/Microsoft.App/stable/2025-07-01/ContainerApps.json

```diff
@@ -0,0 +1,1502 @@
+          },
+          "x-ms-client-flatten": true
+        }
+      }
+    },
+    "ContainerAppCollection": {
+      "description": "Container App collection ARM resource.",
+      "required": [
+        "value"
+      ],
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "Collection of resources.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ContainerApp"
+          }
+        },
+        "nextLink": {
+          "description": "Link to next page of resources.",
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
+    "Dapr": {
+      "description": "Container App Dapr configuration.",
+      "type": "object",
+      "properties": {
+        "enabled": {
+          "description": "Boolean indicating if the Dapr side car is enabled",
+          "type": "boolean",
+          "default": false
+        },
+        "appId": {
+          "description": "Dapr application identifier",
+          "type": "string"
+        },
+        "appProtocol": {
+          "description": "Tells Dapr which protocol your application is using. Valid options are http and grpc. Default is http",
+          "enum": [
+            "http",
+            "grpc"
+          ],
+          "type": "string",
+          "default": "http",
+          "x-ms-enum": {
+            "name": "appProtocol",
+            "modelAsString": true
+          }
+        },
+        "appPort": {
+          "format": "int32",
+          "description": "Tells Dapr which port your application is listening on",
+          "type": "integer"
+        },
+        "httpReadBufferSize": {
+          "type": "integer",
+          "description": "Dapr max size of http header read buffer in KB to handle when sending multi-KB headers. Default is 65KB.",
+          "format": "int32"
+        },
+        "httpMaxRequestSize": {
+          "type": "integer",
+          "description": "Increasing max size of request body http and grpc servers parameter in MB to handle uploading of big files. Default is 4 MB.",
+          "format": "int32"
+        },
+        "logLevel": {
+          "description": "Sets the log level for the Dapr sidecar. Allowed values are debug, info, warn, error. Default is info.",
+          "enum": [
+            "info",
+            "debug",
+            "warn",
+            "error"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "logLevel",
+            "modelAsString": true
+          }
+        },
+        "enableApiLogging": {
+          "description": "Enables API logging for the Dapr sidecar",
+          "type": "boolean"
+        },
+        "appHealth": {
+          "description": "Dapr application health check configuration",
+          "type": "object",
+          "properties": {
+            "enabled": {
+              "description": "Boolean indicating if the health probe is enabled",
+              "type": "boolean"
```

Comment:
>"type": "boolean" [](http://example.com/codeflow?start=14&length=17)

ARM recommends enum over boolean, even for two-state properties, since they document better and can be upgraded to multistate. Consider something like runMode: { Disabled, Active } or better names you choose.